plugins/include/csstats.inc

```diff
@@ -8,62 +8,267 @@
 
 #if defined _csstats_included
-  #endinput
+	#endinput
 #endif
 #define _csstats_included
 
-/* Gets stats from given weapon index. If wpnindex is 0
-* then the stats are from all weapons. If weapon has not been used function
-* returns 0 in other case 1. Fields in stats are:
-* 0 - kills
-* 1 - deaths
-* 2 - headshots
-* 3 - teamkilling
-* 4 - shots
-* 5 - hits
-* 6 - damage
+/**
+ * Retrieves the client's current weapon statistics.
+ *
+ * @note For a list of default CS weapon ids see the CSW_* constants in
+ *       amxconst.inc, this function also works on custom weapons.
+ * @note For a list of possible body hitplaces see the HIT_* constants in
+ *       amxconst.inc
+ * @note The fields in the statistics are:
+ *          0 - Kills
+ *          1 - Deaths
+ *          2 - Headshots
+ *          3 - Teamkills
+ *          4 - Shots
+ *          5 - Hits
+ *          6 - Damage
+ *
+ * @param index     Client index
+ * @param wpnindex  Weapon id, or 0 to retrieve total statistics across all
+ *                  weapons
+ * @param stats     Buffer to copy statistics to
+ * @param bodyhits  Buffer to copy body hits to
+ *
+ * @return          1 on success, 0 if no statistics are available for the weapon
+ *                  id
+ * @error           If an invalid client index or weapon id is provided, an
+ *                  error will be thrown.
+ */
+native get_user_wstats(index, wpnindex, stats[8], bodyhits[8]);
 
-* For body hits fields see amxconst.inc. */
-native get_user_wstats(index,wpnindex,stats[8],bodyhits[8]);
+/**
+ * Retrieves the client's weapon statistics from the current round.
+ *
+ * @note For a list of default CS weapon ids see the CSW_* constants in
+ *       amxconst.inc, this function also works on custom weapons.
+ * @note For a list of possible body hitplaces see the HIT_* constants in
+ *       amxconst.inc
+ * @note The fields in the statistics are:
+ *          0 - Kills
+ *          1 - Deaths
+ *          2 - Headshots
+ *          3 - Teamkills
+ *          4 - Shots
+ *          5 - Hits
+ *          6 - Damage
+ *
+ * @param index     Client index
+ * @param wpnindex  Weapon id, or 0 to retrieve total statistics across all
+ *                  weapons
+ * @param stats     Buffer to copy statistics to
+ * @param bodyhits  Buffer to copy body hits to
+ *
+ * @return          1 on success, 0 if no statistics are available for the
+ *                  weapon id
+ * @error           If an invalid client index or weapon id is provided, an
+ *                  error will be thrown.
+ */
+native get_user_wrstats(index, wpnindex, stats[8], bodyhits[8]);
 
-/* Gets round stats from given weapon index.*/
-native get_user_wrstats(index,wpnindex,stats[8],bodyhits[8]);
+/**
+ * Retrieves the client's weapon statistics from the permanent storage on the
+ * server.
+ *
+ * @note The permanent storage is updated on every respawn or client disconnect.
+ * @note Player rank is determined by the difference of kills to deaths.
```

Comment:
This is kind of imprecise. Position calculation is provided by a custom function you can modify (csstats.sma). This is probably worth to explain that.